exasol_script_languages_developer_sandbox/lib/aws_access.py

```diff
@@ -112,13 +117,145 @@ def delete_stack(self, stack_name: str) -> None:
         cf_client = self._get_aws_client('cloudformation')
         cf_client.delete_stack(StackName=stack_name)
 
+    def describe_stacks(self) -> List[Any]:
+        """
+        This functions uses Boto3 to describe a all cloudformation stack.
+        """
+        logging.debug(f"Running describe_stacks for aws profile {self.aws_profile_for_logging}")
+        cf_client = self._get_aws_client('cloudformation')
+        current_result = cf_client.describe_stacks()
+        result = current_result["Stacks"]
+
+        while "NextToken" in current_result:
+            current_result = cf_client.describe_stacks(NextToken=current_result["NextToken"])
+            result.extend(current_result["Stacks"])
+        return result
+
     def describe_instance(self, instance_id: str):
         """
         Describes an AWS instance identified by parameter instance_id
         """
-        logging.debug(f"Running delete_ec2_key_pair for aws profile {self.aws_profile_for_logging}")
+        logging.debug(f"Running describe_instance for aws profile {self.aws_profile_for_logging}")
+        cloud_client = self._get_aws_client("ec2")["Reservations"][0]["Instances"][0]
+        return cloud_client.describe_instances(InstanceIds=[instance_id])
+
+    def create_image_from_ec2_instance(self, instance_id: str, name: str, tag_value: str, description: str) -> str:
+        """
+        Creates an AMI image from an EC-2 instance
+        """
+        logging.debug(f"Running create_image_from_ec2_instance for aws profile {self.aws_profile_for_logging}")
         cloud_client = self._get_aws_client("ec2")
-        return cloud_client.describe_instances(InstanceIds=[instance_id])["Reservations"][0]["Instances"][0]
+        tags = [{"ResourceType": "image", "Tags": create_default_asset_tag(tag_value)},
```

Comment:
do we need the usual exasol contact tags as well? exa:owner, exa:mail

Reply:
I don't think they will make much sense. In the normal release-flow, it will be the exa-ci4 user who will be the owner of all the asset. For development, I use my user account, so there is also no need to track resources.